docs/use-cases/microfrontends.md

```diff
@@ -0,0 +1,308 @@
+| `checkout`    | Buy products    | `/checkout` | Everyone            |
+| `account`     | Manage account  | `/account`  | Authenticated users |
+
+Each microfrontend will be accessible via its own URL path, like `yoursite.com/products`, `yoursite.com/signup`, etc.
+
+
+{% callout type="note" title="Advanced microfrontends" %}
+We are using a very simple example here. But in a real world application, each microfrontend can be much more advanced, where a microfrontend can be:
+
+- taking over a single route, or
+- taking over a group of routes, or
+- rendering one or more components on a page that is already owned by another microfrontend
+{% /callout %}
+
+## Configuring tags
+
+Your entire application can contain several feature flags, but given it is a microfrontends architecture, we want to make sure that each microfrontend only loads the features it needs.
+
+Otherwise it will result into bloated datafiles with unnecessary data transfer and processing that will slow down your application.
+
+To achieve that, we need to let our Featurevisor [configuration](/docs/configuration) know which tags we want to build our datafiles for:
+
+```js
+// featurevisor.config.js
+module.exports = {
+  environments: [
+    "staging",
+    "production",
+  ],
+
+  tags: [
+    'products',
+    'signup',
+    'signin',
+    'checkout',
+    'account'
+  ]
+};
+```
+
+Once this configuration is in place, we can [build](/docs/building-datafiles) our datafiles:
+
+```
+$ featurevisor build
+```
+
+And it will output the following datafiles in the `dist` directory:
+
+```
+$ tree dist
+dist
+├── production
+│   ├── datafile-tag-products.json
+│   ├── datafile-tag-signup.json
+│   ├── datafile-tag-signin.json
+│   ├── datafile-tag-checkout.json
+│   └── datafile-tag-account.json
+└── staging
+│   ├── datafile-tag-products.json
+│   ├── datafile-tag-signup.json
+│   ├── datafile-tag-signin.json
+│   ├── datafile-tag-checkout.json
+│   └── datafile-tag-account.json
+
+2 directories, 10 files
+```
+
+## Attributes
+
+We will set up some Featurevisor attributes first, that we will use in various stages of this guide later.
+
+### `userId`
+
+```yml
+# attributes/userId.yml
+description: Unique identifier of the logged in User
+type: string
+capture: true
+```
+
+### `deviceId`
+
+This ID can be generated at the client-side level when user first visits your app, and stored in a cookie or localStorage for e.g. if in a browser environment.
+
+```yml
+# attributes/deviceId.yml
+description: Unique identifier of the device
+type: string
+capture: true
+```
+
+## Feature
+
+For this example, we will create a new feature flag that's responsible for toggling a marketing banner that's shown at the bottom of a page. We can call it `showMarketingBanner`.
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - <your-tag-here> # we will discuss this in next section below
+
+defaultVariation: false
```

Comment:
given this is not the focus of this example, is this key needed here? It feels a bit distracting

Reply:
as of this moment, this property `defaultVariation` is required for all features. otherwise linting will fail, and the datafiles won't be generated.

my idea is to start with a fully functional feature in this section, and then talk about its various parts with more focus in later sections.